pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -338,15 +340,115 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
+// It checks whether the deleted pod will conflict with volumes of other pods on the same node
+func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	oldPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPodDeleted", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	if pl.verifyReadWriteOncePodPVCConflict(logger, pod) {
+		return framework.QueueAfterBackoff
+	}
+
+	nodeInfo := framework.NewNodeInfo(oldPod)
+	if !satisfyVolumeConflicts(pod, nodeInfo) {
+		return framework.QueueAfterBackoff
+	}
+
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterNodeAdded is invoked whenever a node added, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterNodeAdded(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	_, newNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterNodeAdded", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	if pl.verifyReadWriteOncePodPVCConflict(logger, pod) {
+		return framework.QueueAfterBackoff
+	}
+
+	logger.V(4).Info("node was added", "pod", klog.KObj(pod), "node", klog.KObj(newNode))
+	return framework.QueueSkip
+}
+
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim changed, It checks whether
+// that change made a previously unschedulable pod schedulable.
+func (pl *VolumeRestrictions) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) framework.QueueingHint {
+	oldPersistentVolumeClaim, _, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterPersistentVolumeClaimChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.QueueAfterBackoff
+	}
+
+	if oldPersistentVolumeClaim == nil {
+		return framework.QueueImmediately
+	}
+
+	// The new pvc won't affect pod scheduling because it hasn't bound any pod yet, and the AccessModes of the pvc are immutable,
+	// so the modified pvc won't affect pod scheduling, and we only need to determine if the current pod has a ReadWriteOncePod
+	// that is bound by other pods to determine whether the pod is schedulable or not
+	if pl.verifyReadWriteOncePodPVCConflict(logger, pod) {
+		return framework.QueueAfterBackoff
+	}
+
+	return framework.QueueSkip
+}
+
+// verifyReadWriteOncePodPVCConflict Checks if the pod's pvc conflicts with other pods, if ReadWriteOncePod is not enabled, it returns false.
+func (pl *VolumeRestrictions) verifyReadWriteOncePodPVCConflict(logger klog.Logger, pod *v1.Pod) bool {
+	if !pl.enableReadWriteOncePod {
+		return false
+	}
```

Comment:
It looks odd from the function name `verifyReadWriteOncePodPVCConflict`. Do we really need to check this here? Can we move this check to outside of this function?